src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -603,17 +605,28 @@ private static ulong CurlReceiveHeadersCallback(IntPtr buffer, ulong size, ulong
                     CurlHandler.EventSourceTrace("Size: {0}", size, easy: easy);
                     try
                     {
-                        // The callback is invoked once per header; multi-line headers get merged into a single line.
-                        string responseHeader = Marshal.PtrToStringAnsi(buffer).Trim();
                         HttpResponseMessage response = easy._responseMessage;
 
-                        if (!TryParseStatusLine(response, responseHeader, easy))
+                        CurlResponseHeaderReader reader = new CurlResponseHeaderReader(buffer, size);
```

Comment:
Thanks for the pointers to the RFCs, David. I referenced RFC 7230 while working on this. The status line and headers chars we're parsing should be ASCII per the RFC. According to RFC 7230, it is possible that a header value has chars outside of ASCII:

> 3.2.4.  Field Parsing
> Historically, HTTP has allowed field content with text in the
   ISO-8859-1 charset [ISO-8859-1], supporting other charsets only
   through use of [RFC2047] encoding.  In practice, most HTTP header
   field values use only a subset of the US-ASCII charset [USASCII].
   Newly defined header fields SHOULD limit their field values to
   US-ASCII octets.  A recipient SHOULD treat other octets in field
   content (obs-text) as opaque data.

 ...but we're not parsing the header values (other than a `Trim` on the edges) and just passing off the header value substring to `Marshal.PtrToStringAnsi` as was being done before this PR.